src/diffusers/pipelines/unidiffuser/pipeline_unidiffuser.py

```diff
@@ -0,0 +1,1025 @@
+    """
+
+    images: Optional[Union[List[PIL.Image.Image], np.ndarray]]
+    text: Optional[Union[List[str], List[List[str]]]]
+
+
+class UniDiffuserPipeline(DiffusionPipeline):
+    r"""
+    Pipeline for a bimodal image-text [UniDiffuser](https://arxiv.org/pdf/2303.06555.pdf) model, which supports
+    unconditional text and image generation, text-conditioned image generation, image-conditioned text generation, and
+    joint image-text generation.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods the
+    library implements for all the pipelines (such as downloading or saving, running on a particular device, etc.)
+
+    Args:
+        vae ([`AutoencoderKL`]):
+            Variational Auto-Encoder (VAE) Model to encode and decode images to and from latent representations. This
+            is part of the UniDiffuser image representation, along with the CLIP vision encoding.
+        text_encoder ([`CLIPTextModel`]):
+            Frozen text-encoder. Similar to Stable Diffusion, UniDiffuser uses the text portion of
+            [CLIP](https://huggingface.co/docs/transformers/model_doc/clip#transformers.CLIPTextModel), specifically
+            the [clip-vit-large-patch14](https://huggingface.co/openai/clip-vit-large-patch14) variant.
+        text_decoder ([`UniDiffuserModel`]):
+            Frozen text decoder. This is a GPT-style model which is used to generate text from the UniDiffuser
+            embedding.
+        image_encoder ([`CLIPVisionModel`]):
+            UniDiffuser uses the vision portion of
+            [CLIP](https://huggingface.co/docs/transformers/model_doc/clip#transformers.CLIPVisionModel) to encode
+            images as part of its image representation, along with the VAE latent representation.
+        tokenizer ([`CLIPTokenizer`]):
+            Tokenizer of class
+            [CLIPTokenizer](https://huggingface.co/docs/transformers/v4.21.0/en/model_doc/clip#transformers.CLIPTokenizer).
+        image_processor ([`CLIPImageProcessor`]):
+            CLIP image process of class
+            [CLIPImageProcessor](https://huggingface.co/docs/transformers/model_doc/clip#transformers.CLIPImageProcessor),
+            used to preprocess the image before CLIP encoding.
+        unet ([`UniDiffuserModel`]):
+            UniDiffuser uses a [U-ViT](https://github.com/baofff/U-ViT) model architecture, which is like a
+            [`Transformer2DModel`] with U-Net-style skip connections. [`UniDiffuserModel`] additionally has input and
+            output heads on top of a base [`UTransformer2DModel`].
+        scheduler ([`SchedulerMixin`]):
+            A scheduler to be used in combination with `unet` to denoise the encoded image latents. Can be one of
+            [`DDIMScheduler`], [`LMSDiscreteScheduler`], or [`PNDMScheduler`].
+    """
+
+    def __init__(
+        self,
+        vae: AutoencoderKL,
```

Comment:
```suggestion
        vae: AutoencoderKL,
        tokenizer: GPT2Tokenizer,
```
we can add the tokenizer directly here

Reply:
Refactored `UniDiffuserPipeline` `__init__` to include `text_tokenizer: GPT2Tokenizer` (as well as `clip_tokenizer: CLIPTokenizer` from before).